### DISCUSSION TITLE
Copy Table between PBIX

### DISCUSSION THREAD

**Discussion by mweschenburg** [TabularEditor/TabularEditor3#334] [🙋♀️TE3 Q&A]
I am a novice user of TE2.  Mostly I have used it to bo bulk edits of calculated measures in Power BI.  I am looking to broaden my use of TE2, and considering TE3 if it accomplishes some of the more tedious development I need to do.

Ideally, I want to be able to copy a table from once PBIX model and duplicate it in another PBIX model.  The table has many fields with formatting, properties, etc; and several hierarchies.  It would be very tedious to replicate all properties manually.

![image](https://user-images.githubusercontent.com/97113383/148084908-ef7316e9-21ca-46a0-b6cf-ed770333ef2b.png)

Specifically, I am trying to use the "Replace Table" feature documented in below screenshot.  I have been mostly unsuccessful except in very limited/controlled instances.  Are there any undocumented constraints?  

My trials failed in even the simplest use case:

1. Create new PBIX model.
2. Import single table.
3. Create hierarchy.
4. Use TE3 (eval version) to copy table.
5. Create another new PBIX model.
6. Use TE3 to paste table.  Table displays in TOM Explorer.
7. Save changes.  Table displays in Power BI Desktop, with apply changes prompt.
8. Table is removed from Power BI Desktop after applying changes.

I was able to overwrite properties of an existing table. 

1. Create new PBIX model.
2. Import single table.
3. Create hierarchy.
4. Use TE3 (eval version) to copy table.
5. Create another new PBIX model.
6. Import same table as in first PBIX.
7. Use TE3 to paste table, overwriting the original.  Table displays in TOM Explorer.
8. Save changes.  Table displays in Power BI Desktop, with apply changes prompt.
9. The hierarchy ad other field properties are applied successfully to the table.

**Comment by otykier** [comment]
Please note that editing/deleting/creating/replacing tables against a model in Power BI Desktop is [not currently supported by Microsoft](https://docs.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#data-modeling-operations). This explains why the table is removed after saving/applying changes in Power BI Desktop. There is no known workaround currently.
